example/provider-extensions/garden/configure-admission.sh

```diff
@@ -0,0 +1,63 @@
+#!/usr/bin/env bash
+#
+# Copyright 2023 SAP SE or an SAP affiliate company. All rights reserved. This file is licensed under the Apache Software License, v. 2 except as noted otherwise in the LICENSE file
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses~LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+set -e
+
+SCRIPT_DIR="$( cd "$( dirname "${BASH_SOURCE[0]}" )" &> /dev/null && pwd )"
+REPO_ROOT_DIR="$(realpath "$SCRIPT_DIR"/../../..)"
+
+usage() {
+  echo "Usage:"
+  echo "> configure-admission.sh [ -h | <garden-kubeconfig> <apply|delete> ]"
+  echo
+  echo ">> For example: configure-admission.sh ~/.kube/garden-kubeconfig.yaml apply"
+
+  exit 0
+}
+
+if [ "$1" == "-h" ] || [ "$#" -ne 2 ]; then
+  usage
+fi
+
+if [ "$2" != "apply" ] && [ "$2" != "delete" ]; then
+  usage
+fi
+
+garden_kubeconfig=$1
+command=$2
+
+for p in $(yq '. | select(.kind == "ControllerDeployment") | select(.metadata.name == "provider-*") | .metadata.name' $REPO_ROOT_DIR/example/provider-extensions/garden/controllerregistrations/* | grep -v -E "^---$"); do
+  echo "Found \"$p\" in $REPO_ROOT_DIR/example/provider-extensions/garden/controllerregistrations. Trying to configure its admission controller..."
+  if PROVIDER_RELEASES=$(curl --fail -s -L -H "Accept: application/vnd.github+json" https://api.github.com/repos/gardener/gardener-extension-$p/releases); then
+    LATEST_RELEASE=$(jq -r '.[].tag_name' <<< $PROVIDER_RELEASES | head -n 1)
```

Comment:
How about using the same version the ControllerDeployment has?
```suggestion
  LATEST_RELEASE=$(p="$p" yq '. | select(.kind == "ControllerDeployment") | select(.metadata.name == strenv(p)) | .providerConfig.values.image.tag' "$REPO_ROOT_DIR"/example/provider-extensions/garden/controllerregistrations/*)
  if curl --fail -s -L -H "Accept: application/vnd.github+json" https://api.github.com/repos/gardener/gardener-extension-"$p"/releases/tags/"$LATEST_RELEASE" >/dev/null; then
```
plus some changes in variable naming and `echo`

Reply:
As discussed offline, it is hard to determine the "correct" version if a custom build is used. The extension setup is used to test custom provider extension releases. Therefore, it becomes cumbersome to determine the "correct" version as it might not be available, i.e. only the provider container image, but not the admission part was pushed or there is no corresponding branch.
This is why we agreed to keep it as is so that a simple heuristic is used. The admission controller is expected to change less frequently meaning that a slight version mismatch should not matter.